### PR TITLE
Pass in counterparty portID, channelID when verifying channel in ChanUpgradeOpen.

### PR DESCRIPTION
**Pull request by DimitrisJim**
## Description

Fixes a bug raised by Luca Joss in the shared ibc-go-rs channel.

Rename to `proofCounterpartyChannel` to align with the rest.

closes: #XXXX

### Commit Message / Changelog Entry

```bash
fix: pass counterparty portid, channelid.
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: modules/core/04-channel/keeper/upgrade.go [hunk 1]
```diff
@@ -319,7 +319,7 @@ func (k Keeper) ChanUpgradeOpen(
 	portID,
 	channelID string,
 	counterpartyChannelState types.State,
-	proofChannel []byte,
+	proofCounterpartyChannel []byte,
```
Comment (damiannolan) [inline modules/core/04-channel/keeper/upgrade.go:322]: I think I'd prefer to keep this as `proofChannel` as we are always verify proofs of a counterparty (its implicit)

File: modules/core/04-channel/keeper/upgrade.go [hunk 2]
```diff
@@ -397,7 +397,14 @@ func (k Keeper) ChanUpgradeOpen(
...
+		ctx,
+		connection,
+		proofHeight, proofCounterpartyChannel,
+		channel.Counterparty.PortId,
+		channel.Counterparty.ChannelId,
+		counterpartyChannel,
+	)
 	if err != nil {
 		return errorsmod.Wrapf(err, "failed to verify counterparty channel, expected counterparty channel state: %s", counterpartyChannel.String())
 	}
```
Comment (damiannolan) [inline modules/core/04-channel/keeper/upgrade.go:410]: nit: should we consolidate the error check to within the scope of the func call like it is in `startFlushUpgrade`

```go
if err := k.connectionKeeper.VerifyChannelState(...); err != nil {
    return errorsmod.Wrapf(err, "...")
}
```

**Review by damiannolan — APPROVED** [review]
Nice one! Just left nits, but thanks for the fix! 🚀

File: modules/core/04-channel/keeper/upgrade.go (hunk 2 above)
Comment (DimitrisJim) [reply modules/core/04-channel/keeper/upgrade.go:410]: ah yea, definitely.

File: modules/core/04-channel/keeper/upgrade.go (hunk 1 above)
Comment (DimitrisJim) [reply modules/core/04-channel/keeper/upgrade.go:322]: I do think `proofChannel` caused my initial mis-step on this. Other handles also prefer explicit. Change later all together if we decide so?

Comment (DimitrisJim) [reply modules/core/04-channel/keeper/upgrade.go:322]: think there's some areas where different approaches have been taken (e.g also how we do connection look-up) deviating from the old handlers, definitely would prefer to hammer those differences out and be consistent. Opened https://github.com/cosmos/ibc-go/issues/4053 to keep track of it and not forget

**Review by colin-axner — APPROVED** [review]
Nice catch!

**Comment by colin-axner** [comment]
We should add a test case that would catch this?

**Comment by DimitrisJim** [comment]
> We should add a test case that would catch this?

I tried. This issue was a result of both chains having the same values for port/channel. It's tricky since we create the mock IBC app using a portID of `mock` and channel ids by default result in `channel-0` for both. Unsure if I'm missing some other option here.

**Review by crodriguezvega — APPROVED** [review]
Thank you for the quick fix, @DimitrisJim!

**Comment by colin-axner** [comment]
> I tried. This issue was a result of both chains having the same values for port/channel. It's tricky since we create the mock IBC app using a portID of mock and channel ids by default result in channel-0 for both. Unsure if I'm missing some other option here.

One quick hack is to make a second path and call `endpoint.ChanOpenInit` on one side. This will generate channel-0 on that end. Then when you create the path for testing, one side will have `channel-0` and the other will have `channel-1`

File: modules/core/04-channel/keeper/upgrade_test.go [hunk 3]
```diff
@@ -712,6 +712,14 @@ func (suite *KeeperTestSuite) TestChanUpgradeOpen() {
 		suite.Run(tc.name, func() {
 			suite.SetupTest()
 
+			// Create an initial path used only to invoke a ChanOpenInit handshake.
+			// This bumps the channel identifier generated for chain A on the
+			// next path used to run the upgrade handshake.
+			// See issue 4062.
```
Comment (DimitrisJim) [inline modules/core/04-channel/keeper/upgrade_test.go:718]: linking #4062

File: modules/core/04-channel/keeper/upgrade_test.go [hunk 4]
```diff
@@ -707,6 +707,14 @@ func (suite *KeeperTestSuite) TestChanUpgradeOpen() {
 		},
 	}
 
+	// Create an initial path used only to invoke a ChanOpenInit handshake.
+	// This bumps the channel identifier generated for chain A on the
+	// next path used to run the upgrade handshake.
+	// See issue 4062.
```
Comment (DimitrisJim) [inline modules/core/04-channel/keeper/upgrade_test.go:713]: linking #4062

**Pull request merged**